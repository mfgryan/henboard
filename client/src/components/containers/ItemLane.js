// redux dep
import { connect } from "react-redux";
import { openInfo, removeItem } from "../../actions/items";
import projects from "../../models/projects.js";
import sprints from "../../models/sprints.js";

// component dep
import Item from "../Item";

const mapStateToProps = (state, ownProps) => {
    let project = projects.getCurrentProject(state);
    let sprint = sprints.getCurrentSprint(state);
    return {
        draggable: !ownProps.openInfo,
        item: {
            project: project.project,
            week: sprint.week,
            column: ownProps.column,
            name: ownProps.name,
            openInfo: ownProps.openInfo,
            value: ownProps.value
        }
    }
};

const mapDispatchToProps = (dispatch, ownProps) => {
    return {
        dragItem: (event,item) => { 
            event.dataTransfer.setData("text", item.name);
        },
        itemClicked: (event,item) => {
            dispatch(openInfo({project: item.project, name: item.name}));
        },
        removeItem: (item) => {
            dispatch(removeItem(item))
        },
    };
};

const ItemLane = connect(  
    mapStateToProps,
    mapDispatchToProps
)(Item);

export default ItemLane;
