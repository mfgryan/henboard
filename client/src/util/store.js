// redux dep
import { createStore, applyMiddleware } from "redux";
import { logger, validate } from "../util/middleware.js";
import data from "../util/data.js";
import rootReducer from "../reducers/index.js";

const store = createStore(rootReducer,applyMiddleware(logger,validate));

export { store }
