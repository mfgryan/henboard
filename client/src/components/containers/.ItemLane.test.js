import React from "react";
import ItemLane from "./ItemLane";
import renderer from "react-test-renderer";

// redux dep
import { Provider } from "react-redux";
import store from "../../util/store.js";

// router dep
import { BrowserRouter as Router } from "react-router-dom";

it("ItemLane renders correctly", () => {
    let item = {
        project: "henboard",
        week: "01/01/01",
        column: "Todo",
        name: "bar",
        openInfo: false,
        value: "foo"
    };
    const tree = renderer
        .create(
            <Router>
                <Provider store={store}>
                    <ItemLane openInfo item={item} />
                </Provider>
            </Router>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
