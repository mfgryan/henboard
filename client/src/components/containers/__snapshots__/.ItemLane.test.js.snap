exports[`test ItemLane renders correctly 1`] = `
<div
  className="Item row"
  draggable={false}
  onDragStart={[Function]}>
  <div
    className="col-md-10">
    <p
      onClick={[Function]} />
  </div>
  <div
    className="col-md-2">
    <p
      onClick={[Function]}>
      X
    </p>
  </div>
</div>
`;
